### PR TITLE
Add checkbox style to remove bogus rendered bullet

### LINKED ISSUE #202: PR for checkbox possible?
## Checklist

- [x] I have read the [contributing guidelines](https://github.com/cotes2020/jekyll-theme-chirpy/blob/master/.github/CONTRIBUTING.md).
- [x] There are no similar request on existing issues (including closed ones).
- [x] I have read the [project progress](https://github.com/cotes2020/jekyll-theme-chirpy/projects) and know the current progress of the project.
- [x] I was in the `master` branch of the latest code.

## Is your feature request related to a problem? Please describe

At the moment if I create an unordered list in my Markdown post with the github markdown syntax for checkboxes (`- [ ]`), the list item bullet is displayed and it is touching the checkbox. It's not very pleasing on the eye. :-)

## Describe the solution you'd like

I would like to be able to write blog posts and be able to use the github markdown syntax for checkboxes (`- [ ]` or `* [ ]` for todo and `- [x]` or `* [x]` for done). 

## Describe alternatives you've considered

I would like to submit a pull request to implement this solution if this is OK with you. It consists in:
* adding a checkbox add-on style in `assets/css/_addon`
* modifying the post layout to replace `<li>[ ]` and `<li>[x]` items in the `post-content` div with the checkbox style defined in the previous file
* modifying `main.scss` in `assets/_addon` to import the new checkbox add-on style

## Additional context

If it's OK with you I can submit the PR.

### PR DESCRIPTION
**Pull request by alexandra-zaharia**
## Description

Fixes #202 (Github-sytle checkbox)

A style for checkboxes is defined in an add-on SCSS file and it is imported in the main SCSS file. The new style is used in the post layout in order to replace `<li>[ ]` and `<li>[x]` items with the new style of checkbox:
* the space between the checkbox and the text is increased
* the `<li>` bullet is suppressed

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How has this been tested

- [x] I have run `bash ./tools/test.sh --build` (at the root of the project) locally and passed
- [x] I have tested this feature in the browser

### Test Configuration

- Browser type & version: Google Chrome Version 87.0.4280.88 (Official Build) (64-bit)
- Operating system: Manjaro Linux (kernel 5.9.11-3-MANJARO)
- Bundler version: 2.1.4
- Ruby version: 2.7.2p137 (2020-10-01 revision 5445e04352) [x86_64-linux]
- Jekyll version: 4.1.1

### Checklist

- [ ] My code follows the [Google style guidelines](https://google.github.io/styleguide/)
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by cotes2020 — CHANGES_REQUESTED** [review]
This solution is too complicated. In fact, we don't need to modify the HTML structure, as long as we can achieve the same style through pure SCSS code:

```scss
  ul {
    padding-left: 2rem;

    .task-list-item {
      list-style-type: none;
    }

    input[type=checkbox] {
      margin: 0 .5rem .25rem -1.3rem;
      vertical-align: middle;
    }
  }
```

Just add above codes before [the closing `.post-content` class](https://github.com/cotes2020/jekyll-theme-chirpy/blob/417b0d3ad960db2ff93b62d88fa769cace78e3b9/assets/css/_addon/main.scss#L863).

**Comment by alexandra-zaharia** [comment]
Alright, thank you for these valuable explanations. I will try to incorporate the requested changes.

File: assets/css/_addon/main.scss [hunk 1]
```diff
@@ -6,6 +6,7 @@
 * MIT Licensed
 */
 
+@import "_addon/checkbox";
```
Comment (cotes2020) [inline assets/css/_addon/main.scss:9]: This line should be deleted

File: assets/css/_addon/checkbox.scss [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+}
+
+.task-list-item input {
+  margin: 0 0.2em 0.25em -1.6em;
+  vertical-align: middle;
+}
+
+.task-list-item + .task-list-item {
+  margin-top: 3px;
+}
```
Comment (cotes2020) [inline assets/css/_addon/checkbox.scss:14]: For now, the new file `checkbox.scss` is unused and should be remove from the PR

**Review by cotes2020 — CHANGES_REQUESTED** [review]

File: assets/css/_addon/checkbox.scss (hunk 2 above)
Comment (alexandra-zaharia) [reply assets/css/_addon/checkbox.scss]: Whoops, indeed!

**Pull request merged**